core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/SimplifyExpressions.java

```diff
@@ -31,16 +33,15 @@
 public class SimplifyExpressions
         extends ExpressionRewriteRuleSet
 {
-    public static Expression rewrite(Expression expression, Session session, PlannerContext plannerContext)
+    public static Expression rewrite(Expression expression, Session session, IrExpressionOptimizer optimizer)
     {
-        requireNonNull(plannerContext, "plannerContext is null");
         if (expression instanceof Reference) {
             return expression;
         }
         expression = pushDownNegations(expression);
         expression = extractCommonPredicates(expression);
         expression = normalizeOrExpression(expression);
-        return new IrExpressionInterpreter(expression, plannerContext, session).optimize();
```

Comment:
Can we have `pushDownNegations`, `extractCommonPredicates`, `normalizeOrExpression` as rules?

Reply:
Certainly, but not as part of this PR.